discord/ext/ipc/client.py

```diff
@@ -11,87 +11,78 @@
     limitations under the License.
 """
 
-
 import asyncio
 import json
-import socket
+import typing
+import aiohttp
 
 from .errors import *
 
 
-class Node:
-    """Base Node class returned when an IPC server is discovered."""
-    
-    def __init__(self, client, data):
-        self.client = client
-        self._json = data
-        
-        self.ip = data.get("multicast_grp")
-        self.port = data.get("port")
-        self.endpoints = data.get("endpoints")
-    
-    def get_json(self):
-        """Convert object to json"""
-        return self._json
-
-    async def request(self, endpoint, **kwargs):
-        if endpoint not in self.endpoints.keys():
-            raise NoEndpointFoundError("Endpoint \"{}\" not found".format(endpoint))
-        
-        return await self.client.request(endpoint, self.port, **kwargs)
-    
-    def __str__(self):
-        return str(self.get_json())
-
 class Client:
-    """Main client class, used for delivering data from the web server to the bot"""
+    """Handles webserver side requests to the bot process.
+
+    :param host: The IP or host of the IPC server, defaults to localhost
+    :type host: ``str``, optional
+    :param port: The port of the IPC server. If not supplied the port will be found automatically, defaults to None
+    :type port: ``int``, optional
+    :param secret_key: The secret key for your IPC server. Must match the server secret_key or requests will not go ahead, defaults to None
+    :type secret_key: ``Union[str, bytes]``, optional
+    """
+
+    def __init__(self, host: str = "localhost", port: int = None, secret_key: typing.Union[str, bytes] = None):
+        """Constructor"""
+        self.loop = asyncio.get_event_loop() or asyncio.new_event_loop()
```

Comment:
You don't need the or here https://github.com/python/cpython/blob/master/Lib/asyncio/events.py#L752-L765

Reply:
I see, thank you for pointing this out.